File: 202.happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        history = {n}
        while n != 1:
            n = sum(list(map(lambda x : int(x) ** 2, str(n))))
            if n in history:
                return False
            history.add(n)
            
        else:
            return True
            

File: 202.happy-number.go
func squareSum(num int) int {
    sum := 0
    for num != 0 {
        remainder := num % 10
        sum = sum + remainder * remainder
        num = num / 10
    }
    return sum
}


func isHappy(n int) bool {
    slow := squareSum(n)
    fast := squareSum(n)
    fast = squareSum(fast)
    for fast != slow {
        slow = squareSum(slow)
        fast = squareSum(fast)
        fast = squareSum(fast)
    }
    return slow == 1
}
